app/jobs/reports/combined_invoice_supplement_report_v2.rb

```diff
@@ -144,8 +144,8 @@ def combine_by_iaa_month(
               iaa_results = by_iaa_and_year_month[ [iaa_key, year_month] ]
               issuer_results = year_months_data[year_month]
               year_month_start = Date.strptime(year_month, '%Y%m')
-              iaa_start_date = Date.parse(iaa_results.first[:iaa_start_date])
-              iaa_end_date = Date.parse(iaa_results.first[:iaa_end_date])
+              iaa_start_date = iaa_results ? Date.parse(iaa_results.first[:iaa_start_date]) : 'n/a'
+              iaa_end_date = iaa_results ? Date.parse(iaa_results.first[:iaa_end_date]) : 'n/a'
 
```

Comment:
There are a few different troubleshooting options.

One would be to make sure all the lists used in the `combine_by_iaa_month` have been compacted.
Another would be doing a `next unless iaa_results`

I've opted to let the iteration continue so that we can get more visibility into why the report would be failing in the first place. All of the touch points along the way seem to be around the `iaas` but in the prod console none of them failed to have data. There may be an issue deeper in the monthHelper but this should at least help us see which Iaas_months have n/a dates in the report.

Reply:
if `iaa_results` was an empty array this would still error, right?

Reply:
Yes, there a a bunch of different ways it can fail. From first not having a key, or a poorly written key which would end up being ` undefined '[]' for nil` and eventually `Date.parse Invalid date`. There is the seldom used `rescue "n/a"` but I think something like `iaa_results.first&.dig(:iaa_end_date)` is cleaner, you'll just either nave nil or 'n/a'